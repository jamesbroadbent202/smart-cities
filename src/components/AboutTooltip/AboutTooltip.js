import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@atlaskit/tooltip';
import questionIcon from './Question.svg';
import style from './AboutTooltip.scss';

const AboutTooltip = props => (
  <Tooltip
    description={props.description}
    position={props.position}
  >
    <img
      className={style.icon}
      width={props.size}
      height={props.size}
      src={questionIcon}
      alt="about this indicator"
    />
  </Tooltip>
);

AboutTooltip.propTypes = {
  description: PropTypes.string.isRequired,
  position: PropTypes.oneOf([ // defined by atlaskit/tooltip
    'left',
    'right',
    'top',
    'bottom',
  ]),
  size: PropTypes.number,
};

AboutTooltip.defaultProps = {
  position: 'left',
  size: 16,
};

export default AboutTooltip;
