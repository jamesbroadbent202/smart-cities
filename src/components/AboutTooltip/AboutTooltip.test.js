import React from 'react';
import { shallow } from 'enzyme';
import AboutTooltip from './AboutTooltip';

const defaultProps = {
  description: 'This is the content of the tooltip',
};

it('should match Snapshot', () => {
  const component = shallow(
    <AboutTooltip {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should pass descriptions and position to the tooltip component', () => {
  const component = shallow(
    <AboutTooltip
      description="the description"
      position="right"
    />
  );

  const tooltipProps = component.find('Tooltip').props();

  expect(tooltipProps).toEqual(expect.objectContaining({
    description: 'the description',
    position: 'right',
  }));
});
