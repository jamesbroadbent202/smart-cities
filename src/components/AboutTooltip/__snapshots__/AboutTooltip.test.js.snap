// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<Tooltip description=\\"This is the content of the tooltip\\" position=\\"left\\">
  <img className=\\"icon\\" width={16} height={16} src=\\"Question.svg\\" alt=\\"about this indicator\\" />
</Tooltip>"
`;
